Mocha 6 fails with pnpm: Cannot find module 'semver'

<!--
Have you read Mocha's Code of Conduct? By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/mochajs/mocha/blob/master/.github/CODE_OF_CONDUCT.md
For more, check out the Mocha Gitter chat room: https://gitter.im/mochajs/mocha

Detail the steps necessary to reproduce the problem. To get the fastest support, create an MCVE and upload it to GitHub.
create an [MCVE](https://stackoverflow.com/help/mcve) and upload it to GitHub.
-->

### Prerequisites

<!--
Place an `x` between the square brackets on the lines below for every satisfied prerequisite.
-->

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [x] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend that you _not_ install Mocha globally.

### Description

mocha 6 fails to run under pnpm

### Steps to Reproduce

1. `pnpm install mocha`
2. `pnpx mocha --version` or `npx mocha --version`

**Expected behavior:**

`6.1.2` is displayed in the console

**Actual behavior:**

`Cannot find module 'semver'` is displayed in the console

**Reproduces how often:**

100%

### Versions

6.1.2

<!-- If applicable, please specify: -->

- The output of `mocha --version` and `node node_modules/.bin/mocha --version`:
**`Cannot find module 'semver'`**
- The output of `node --version`:
**`v8.15.1`**
- Your operating system
  - name and version: **Ubuntu 14, Windows 10**
  - architecture (32 or 64-bit): **64, 64**
- Your shell (e.g., bash, zsh, PowerShell, cmd): **bash, Powershell, git bash**
- Your browser and version (if running browser tests):  **N/A**
- Any third-party Mocha-related modules (and their versions):  **N/A**
- Any code transpiler (e.g., TypeScript, CoffeeScript, Babel) being used (and its version):  **N/A**

### Additional Information

A mocha dependency, `node-environment-flags`, requires semver, but does not list it as a dependency in package.json.  See issue: boneskull/node-environment-flags#2 and pull request:  boneskull/node-environment-flags#4.

--- a/package.json
+++ b/package.json
@@ -516,7 +516,7 @@
     "minimatch": "3.0.4",
     "mkdirp": "0.5.1",
     "ms": "2.1.1",
-    "node-environment-flags": "1.0.4",
+    "node-environment-flags": "1.0.5",
     "object.assign": "4.1.0",
     "strip-json-comments": "2.0.1",
     "supports-color": "6.0.0",

